YoutubeExplode/Videos/Streams/StreamClient.cs

```diff
@@ -32,11 +32,32 @@ public StreamClient(HttpClient http)
         _controller = new StreamController(http);
     }
 
+    private async Task<string?> TryGetPlayerSource()
+    {
+        //Use iframe api to get player version
+        var iframeContent = await _http.GetStringAsync("https://www.youtube.com/iframe_api");
+        
+        var version = Regex.Match(iframeContent, @"player\\?/([0-9a-fA-F]{8})\\?/");
+        if (!version.Success) return null;
+
+        //Download player and return it
+        var source = await _http.GetStringAsync($"https://www.youtube.com/s/player/{version.Groups[1]}/player_ias.vflset/en_US/base.js");
+        return source;
```

Comment:
Are you sure you will always get the same player here and on the watch page?

Reply:
this is the way yt-dlp gets the player url

https://github.com/yt-dlp/yt-dlp/blob/c26f9b991a0681fd3ea548d535919cec1fbbd430/yt_dlp/extractor/youtube.py#L2658-L2666